spec/models/ems_refresh/parsers/kubernetes_spec.rb

```diff
@@ -94,4 +95,169 @@
+        :common_partition      => nil
+      },
+      {
+        :volume                => RecursiveOpenStruct.new(:name => "example-volume7",
+                                                          :rbd  => {:fsType   => "user-fs-type",
+                                                                    :readOnly => false}),
+        :name                  => "example-volume7",
+        :git_repository        => nil,
+        :empty_dir_medium_type => nil,
+        :gce_pd_name           => nil,
+        :common_fs_type        => "user-fs-type",
+        :common_path           => nil,
+        :common_read_only      => false,
+        :common_secret         => nil,
+        :common_volume_id      => nil,
+        :common_partition      => nil
+      },
+      {
+        :volume                => RecursiveOpenStruct.new(:name   => "example-volume8",
+                                                          :secret => {:secretName => "example-secret"}),
+        :name                  => "example-volume8",
+        :git_repository        => nil,
+        :empty_dir_medium_type => nil,
+        :gce_pd_name           => nil,
+        :common_fs_type        => nil,
+        :common_path           => nil,
+        :common_read_only      => nil,
+        :common_secret         => "example-secret",
+        :common_volume_id      => nil,
+        :common_partition      => nil
+      },
+      {
+        :volume                => RecursiveOpenStruct.new(:name   => "example-volume9",
+                                                          :cinder => {:volumeId => "example-id"}),
+        :name                  => "example-volume9",
+        :git_repository        => nil,
+        :empty_dir_medium_type => nil,
+        :gce_pd_name           => nil,
+        :common_fs_type        => nil,
+        :common_path           => nil,
+        :common_read_only      => nil,
+        :common_secret         => nil,
+        :common_volume_id      => "example-id",
+        :common_partition      => nil
+      },
+      {
+        :volume                => RecursiveOpenStruct.new(:name              => "example-volume10",
+                                                          :gcePersistentDisk => {:partition => "default-partition"}),
+        :name                  => "example-volume10",
+        :git_repository        => nil,
+        :empty_dir_medium_type => nil,
+        :gce_pd_name           => nil,
+        :common_fs_type        => nil,
+        :common_path           => nil,
+        :common_read_only      => nil,
+        :common_secret         => nil,
+        :common_volume_id      => nil,
+        :common_partition      => "default-partition"
+      }
+    ]
+
+    it "tests example volumes" do
+      parsed_volumes = parser.send(:parse_volumes, example_volumes.collect { |ex| ex[:volume] })
+
+      example_volumes.zip(parsed_volumes).each do |example, parsed|
+        parsed[:name].should be == example[:name]
```

Comment:
@zakiva it would probably more clear to do:

    parsed.should have_attributes(
        :name           => example[:name],
        :git_repository => example[:git_repository],
        ...
    )

Check other tests, `have_attributes` is widely used.